src/App.tsx

```diff
@@ -1,156 +1,207 @@
+import { UserData } from './types/userData';
+import { Todo } from './types/Todo';
+import {
+  deleteTodoOnServer,
+  getTodosFromServer,
+  setTodoCompleteStatus,
+} from './api';
+import { TodoItem } from './components/Todo';
+import { TodoHeader } from './components/TodoHeader';
+import { Footer } from './components/Footer';
+import { Notification } from './components/Notification';
+
+const defaultUser = {
+  createdAt: '2023-07-10T13:09:53.578Z',
+  email: 'gookidoo@gmail.com',
+  id: 11038,
+  name: 'Віктор Булденко',
+  phone: null,
+  updatedAt: '2023-07-10T13:09:53.578Z',
+  username: null,
+  website: null,
+};
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [step, setStep] = useState(LogingSteps.EMAIL);
+  const [user, setUser] = useState<UserData>(defaultUser);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [uncomplete, setUncomplete] = useState<number>(0);
+  const [showFooter, setShowFooter] = useState<boolean>(false);
+  const [sortTodosBy, setSortTodosBy] = useState<Etodos>(Etodos.ALL);
+  const [respError, setRespError] = useState<ResponseError>(ResponseError.NOT);
+  const [toggleActiveTodos, setToggleActiveTodos] = useState(true);
+
+  const toggleTodosActive = () => {
+    const promiseList = todos.map(todo => {
+      if (todo.completed !== toggleActiveTodos) {
+        return setTodoCompleteStatus(todo.id, { completed: toggleActiveTodos });
+      }
+
+      return [];
+    });
+
+    setToggleActiveTodos(!toggleActiveTodos);
+
+    Promise.all(promiseList)
+      .then(() => {
+        getTodosFromServer(user.id).then((todoList) => {
+          setTodos(todoList);
+          setShowFooter(Boolean(todoList.length));
+        });
+      });
+  };
+
+  const checkCompletedTodo = (arr: Todo[]) => {
+    let counter = 0;
+
+    for (let i = 0; i < arr.length; i += 1) {
+      if (!arr[i].completed) {
+        counter += 1;
+      }
+    }
+
+    setUncomplete(counter);
+  };
+
+  const deleteTodo = (id: number) => {
+    deleteTodoOnServer(id).then(() => {
+      getTodosFromServer(user.id).then((todoList) => {
+        setTodos(todoList);
+        checkCompletedTodo(todoList);
+        setShowFooter(Boolean(todoList.length));
+      });
+    }).catch(() => setRespError(ResponseError.ADD));
+  };
+
+  const updateTodo = (todoId: number, obj: Partial<Todo>) => {
+    setTodoCompleteStatus(todoId, obj).then(() => {
+      getTodosFromServer(user.id).then((todoList) => {
+        setTodos(todoList);
+        checkCompletedTodo(todoList);
+        setShowFooter(Boolean(todoList.length));
+      });
+    }).catch(() => setRespError(ResponseError.UPDATE));
+  };
+
+  const displayTodos = (sortBy: Etodos) => {
+    const BASE_URL = 'https://mate.academy/students-api';
+    let endpoint = '';
+
+    const deleteCompleted = () => {
+      fetch(`${BASE_URL}/todos?userId=${user.id}&completed=true`)
+        .then((data) => data.json())
```

Comment:
Such queries better move to separate file `api.js`. Fix all cases

Reply:
rework this case -> 'getTodosFromServer(user.id, 'completed=true')'